addons/web/static/tests/fields/field_utils_tests.js

```diff
@@ -413,4 +413,22 @@ QUnit.test('parse datetime without separator', function (assert) {
     core._t.database.parameters = originalParameters;
 });
 });
+
+QUnit.test('parse smart date input', function (assert) {
+    assert.expect(10);
+
+    const format = "DD MM YYYY";
+    assert.strictEqual(fieldUtils.parse.date("+1d").format(format), moment().add(1, 'days').format(format));
+    assert.strictEqual(fieldUtils.parse.datetime("+2w").format(format), moment().add(2, 'weeks').format(format));
+    assert.strictEqual(fieldUtils.parse.date("+3m").format(format), moment().add(3, 'months').format(format));
+    assert.strictEqual(fieldUtils.parse.datetime("+4y").format(format), moment().add(4, 'years').format(format));
+
+    assert.strictEqual(fieldUtils.parse.date("+5").format(format), moment().add(5, 'days').format(format));
+    assert.strictEqual(fieldUtils.parse.datetime("-5").format(format), moment().subtract(5, 'days').format(format));
+
+    assert.strictEqual(fieldUtils.parse.date("-4y").format(format), moment().subtract(4, 'years').format(format));
+    assert.strictEqual(fieldUtils.parse.datetime("-3m").format(format), moment().subtract(3, 'months').format(format));
+    assert.strictEqual(fieldUtils.parse.date("-2w").format(format), moment().subtract(2, 'weeks').format(format));
+    assert.strictEqual(fieldUtils.parse.datetime("-1d").format(format), moment().subtract(1, 'days').format(format));
```

Comment:
maybe add a few tests with options UTC and timezone?

Reply:
Actually I'm wondering whether those UTC and timezone options are relevant for this function. The parse utils are used in at least two situations:
 1) we received a (textual) value from the server after, eg. a read rpc, and we want to parse it to create a moment
 2) we extracted the value from a date field (so the user just typed something or selected a date in the datepicker), and we want to parse it to notify the change to the model

I think that UTC is only used by the first case, so it is probably useless here. I'm not sure about timezone, but it would be nice to get rid of those options in your new function.